rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/impl/TreeSearch.java

```diff
@@ -0,0 +1,124 @@
+package org.rascalmpl.vscode.lsp.util.locations.impl;
+
+import org.rascalmpl.values.IRascalValueFactory;
+import org.rascalmpl.values.parsetrees.ITree;
+import org.rascalmpl.values.parsetrees.TreeAdapter;
+
+import io.usethesource.vallang.IList;
+import io.usethesource.vallang.IListWriter;
+import io.usethesource.vallang.ISourceLocation;
+import io.usethesource.vallang.IValue;
+
+/**
+ * Utilities for finding sub-trees based on UTF-32 line/column indexing.
+ */
+public class TreeSearch {
+    
+    /**
+     * Helper to test if a line/column position is within the scope of a source location's 
+     * begin line/column and end line/column scope. The first and the last lines are
+     * where the column information is used.
+     * 
+     * @param loc     location with line/column information that acts as a range
+     * @param line    line to check if its within the range
+     * @param column  column to see if we are within the range
+     * @return true iff the given `loc` range spans around the line/column position, inclusively.
+     */
+    private static boolean inside(ISourceLocation loc, int line, int column) {
+        if (!loc.hasLineColumn()) {
+            return false;
+        }
+
+        if (line < loc.getBeginLine() || line > loc.getEndLine()) {
+            return false;
+        }
+
+        if (line == loc.getBeginLine()) {
+            // then make sure we are not before the start
+            return loc.getBeginColumn() <= column;
```

Comment:
what if `endLine == line`, in that case you also have to make sure that `endColumn` is after `column.

Reply:
that's exactly what is checked on line 41.

Reply:
I think there still is a bug. Since we never get to line 43 due to the return on this line. If the loc ends before the column (but is on the same line) it's not inside of it, but this clause would say, as long as it begins before column we're good.

Reply:
ok I'll have a look. thanks

Reply:
you're right! Fixed it.